pkg/broker/filter/filter_handler.go

```diff
@@ -333,15 +334,84 @@ func (h *Handler) getTrigger(ref path.NamespacedNameUID) (*eventingv1.Trigger, e
 	}
-	var filters eventfilter.Filters
-	if filter.Attributes != nil && len(filter.Attributes) != 0 {
-		filters = append(filters, attributes.NewAttributesFilter(filter.Attributes))
+	return eventfilter.NoFilter
+}
+
+func applySubscriptionsAPIFilters(ctx context.Context, filters []eventingv1.SubscriptionsAPIFilter, event cloudevents.Event) eventfilter.FilterResult {
+	return subscriptionsapi.NewAllFilter(materializeFiltersList(ctx, filters)...).Filter(ctx, event)
+}
+
+func materializeSubscriptionsAPIFilter(ctx context.Context, filter eventingv1.SubscriptionsAPIFilter) eventfilter.Filter {
+	var materializedFilter eventfilter.Filter
+	var err error
+	if len(filter.Exact) > 0 {
+		// The webhook validates that this map has only a single key:value pair.
+		for attribute, value := range filter.Exact {
+			materializedFilter, err = subscriptionsapi.NewExactFilter(attribute, value)
+			if err != nil {
+				logging.FromContext(ctx).Debugw("Invalid exact expression", zap.String("attribute", attribute), zap.String("value", value), zap.Error(err))
+				return nil
+			}
+		}
+	}
+	if len(filter.Prefix) > 0 {
+		// The webhook validates that this map has only a single key:value pair.
+		for attribute, prefix := range filter.Exact {
+			materializedFilter, err = subscriptionsapi.NewPrefixFilter(attribute, prefix)
+			if err != nil {
+				logging.FromContext(ctx).Debugw("Invalid prefix expression", zap.String("attribute", attribute), zap.String("prefix", prefix), zap.Error(err))
+				return nil
+			}
+		}
+	}
+	if len(filter.Suffix) > 0 {
+		// The webhook validates that this map has only a single key:value pair.
+		for attribute, suffix := range filter.Exact {
+			materializedFilter, err = subscriptionsapi.NewSuffixFilter(attribute, suffix)
+			if err != nil {
+				logging.FromContext(ctx).Debugw("Invalid suffix expression", zap.String("attribute", attribute), zap.String("suffix", suffix), zap.Error(err))
+				return nil
+			}
+		}
+	}
+	if len(filter.All) > 0 {
+		materializedFilter = subscriptionsapi.NewAllFilter(materializeFiltersList(ctx, filter.All)...)
 	}
-	return filters.Filter(ctx, event)
+	if len(filter.Any) > 0 {
+		materializedFilter = subscriptionsapi.NewAnyFilter(materializeFiltersList(ctx, filter.All)...)
+
+	}
+	if filter.Not != nil {
+		materializedFilter = subscriptionsapi.NewNotFilter(materializeSubscriptionsAPIFilter(ctx, *filter.Not))
+	}
+	if filter.SQL != "" {
+		f, err := subscriptionsapi.NewCESQLFilter(filter.SQL)
+		if err != nil {
+			// This is weird, CESQL expression should be validated when Trigger's are created.
+			logging.FromContext(ctx).Debugw("Found an Invalid CE SQL expression", zap.String("expression", filter.SQL))
+			return nil
+		}
+		materializedFilter = f
+	}
+	return materializedFilter
+}
+
+func materializeFiltersList(ctx context.Context, filters []eventingv1.SubscriptionsAPIFilter) []eventfilter.Filter {
+	materializedFilters := make([]eventfilter.Filter, len(filters))
```

Comment:
This needs 0 as starting index:
```suggestion
	materializedFilters := make([]eventfilter.Filter, 0, len(filters))
```